WMF Framework/Widget/WidgetController.swift

```diff
@@ -27,6 +28,14 @@ public final class WidgetController: NSObject {
         guard #available(iOS 14.0, *), !Bundle.main.isAppExtension else {
             return
         }
+
+        let dataStore = MWKDataStore.shared()
```

Comment:
I wonder if we should find a way to divorce this widget from `MWKDataStore` entirely, to make it more standalone. The `shared` singleton will trigger it's `init`, which will run through setting up the core data stack and such. It seems like there could still be a chance that we get those database lock crashes by using this singleton even though we aren't actually fetching anything from the database. We also wouldn't know if it is still causing those crashes or not, since crashes are only under the umbrella "WidgetsExtension" binary.

Reply:
In this particular case of `let dataStore = MWKDataStore.shared()` we're actually safe though, right? The Bundle check in the guard above will prevent any widget extensions from executing the remainder of `reloadAllWidgetsIfNecessary()`.
```swift
guard #available(iOS 14.0, *), !Bundle.main.isAppExtension else {
```

So for this particular featured article widget, there's no app extension usage of Core Data (unless I'm missing something). We can't further remove all this other Core Data stuff yet because of the other widgets, but at least there's some level of confidence that if the user's only using the featured article widget, there aren't any shared Core Data accesses from that running widget extension, but only from the main app itself.

Reply:
@staykids agh you're right, thank you for pointing that out! Correct, that was the only area that looked like `MWKDataStore` usage from the featured article widget, so I think you're good here.